srtp/srtp.c

```diff
@@ -1149,7 +1439,7 @@ srtp_unprotect_aead (srtp_ctx_t *ctx, srtp_stream_ctx_t *stream, int delta,
      */
     enc_start = (uint32_t*)hdr + uint32s_in_rtp_header + hdr->cc;
     if (hdr->x == 1) {
-        srtp_hdr_xtnd_t *xtn_hdr = (srtp_hdr_xtnd_t*)enc_start;
+        xtn_hdr = (srtp_hdr_xtnd_t*)enc_start;
         enc_start += (ntohs(xtn_hdr->length) + 1);
```

Comment:
I'm worrying about a buffer overflow again here.  Wondering if we need to add some logic to check the packet length from the UDP layer (*pkt_octet_len) against the implied length we derive based on uint32s_in_rtp_header, hdr->cc, and now xtn_hdr->length.